src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -215,6 +216,128 @@ def add_imaging_plane(nwbfile: NWBFile, metadata: dict, imaging_plane_index: int
+    nwbfile : NWBFile
+        The nwbfile to add the plane segmentation to.
+    metadata : dict, optional
+        The metadata for the plane segmentation.
+    plane_segmentation_index: int, optional
+        The index of the plane segmentation to add.
+    image_segmentation_name : str, optional
+        The name of the image segmentation to add the plane segmentation to.
+    iterator_options : dict, optional
+        The options to use when iterating over the image masks of the segmentation extractor.
+    compression_options : dict, optional
+        The options to use when compressing the image masks of the segmentation extractor.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the plane segmentation added.
+    """
+    if image_segmentation_name is None:
+        image_segmentation_name = "ImageSegmentation"
+    if iterator_options is None:
+        iterator_options = dict()
+    if compression_options is None:
+        compression_options = dict()
+
+    def image_mask_iterator():
+        for roi_id in segmentation_extractor.get_roi_ids():
+            image_masks = segmentation_extractor.get_roi_image_masks(roi_ids=[roi_id]).T.squeeze()
+            yield image_masks
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    image_segmentation_metadata = metadata_copy["Ophys"]["ImageSegmentation"]
+    plane_segmentation_metadata = image_segmentation_metadata["plane_segmentations"][plane_segmentation_index]
+    plane_segmentation_name = plane_segmentation_metadata["name"]
+
+    # Check if "ophys" processing module already exists in the NWB file
+    if "ophys" not in nwbfile.processing:
+        ophys = nwbfile.create_processing_module("ophys", "contains optical physiology processed data")
+    else:
+        ophys = nwbfile.get_processing_module("ophys")
+
+    # Check if the image segmentation already exists in the NWB file
+    if image_segmentation_name not in ophys.data_interfaces:
+        ophys.add(ImageSegmentation(image_segmentation_name, "contains image segmentation data"))
+
+    image_segmentation = ophys.get_data_interface(image_segmentation_name)
+
+    # Check if the plane segmentation already exists in the image segmentation
+    if plane_segmentation_name not in image_segmentation.plane_segmentations:
+        roi_ids = segmentation_extractor.get_roi_ids()
+        accepted_ids = [int(roi_id in segmentation_extractor.get_accepted_list()) for roi_id in roi_ids]
+        rejected_ids = [int(roi_id in segmentation_extractor.get_rejected_list()) for roi_id in roi_ids]
+
+        roi_locations = np.array(segmentation_extractor.get_roi_locations()).T
+
+        imaging_plane_metadata = metadata_copy["Ophys"]["ImagingPlane"][plane_segmentation_index]
```

Comment:
You can call `add_imaging_plane` at the beginning of the function (same as you do with the processing module) which will take care of creating the function if it does not exist and then just extract it. This in place of lines between 295-301 here.

Reply:
Neat.